src/components/companies/detail/history/EmissionsLineChart.tsx

```diff
@@ -55,33 +55,53 @@ export default function EmissionsLineChart({
   const endYear = 2030;
   const currentYear = new Date().getFullYear();
 
-  const approximatedData = useMemo(() => {
-    if (dataView !== "overview") {
-      return null;
-    }
-
-    const validPoints = data
+  // Helper to get last two periods with emissions
+  function getLastTwoEmissionsPoints(data: ChartData[]) {
+    return data
       .filter(
         (d): d is ChartData & { total: number } =>
           d.total !== undefined && d.total !== null,
       )
-      .map((d) => ({
-        x: d.year,
-        y: d.total,
-      }))
+      .map((d) => ({ x: d.year, y: d.total }))
       .slice(-2);
+  }
+
+  const approximatedData = useMemo(() => {
+    if (dataView !== "overview") {
+      return null;
+    }
+
+    const regressionPoints: { x: number; y: number }[] = (() => {
+      if (companyBaseYear) {
+        const baseYearPoints = data
+          .filter(
+            (d): d is ChartData & { total: number } =>
```

Comment:
This might be a me-thing more than anything but was unused to see an inlined type guard. I'd personally extract this to a static function instead of defining it inline as it will make the logic of `regressionPoints` easier to read as well.